ccp-openshift-scan/scanning/lib.py

```diff
@@ -0,0 +1,36 @@
+# lib.py containing library functions
+
+import subprocess
+
+
+def run_cmd(cmd, shell=False):
+    """
+    Runs a shell command.
+
+    :param cmd: Command to run
+    :param shell: Whether to run raw shell commands with '|' and redirections
+    :type cmd: str
+    :type shell: boolean
+
+    :return: Command output
+    :rtype: str
+    :raises: subprocess.CalledProcessError
+    """
+    if shell:
+        return subprocess.check_output(cmd, shell=True)
+    else:
+        return subprocess.check_output(cmd.split(), shell=False)
+
+
+def run_cmd_out_err(cmd):
```

Comment:
The places where operations are prone to the exception, it's being handled, for eg `npm` or `gem` or `pip` are not installed.

I think we should not make the build fail if scanner execution fails. That's the old arch behavior.

Reply:
Well, if the scan fails, I do think the build must be considered failure because as far as the service is concerned, only Dockerfile linting is not working directly on the image. All other stages work on the image and if any of them fails, we should consider the build failed, IMO.

Reply:
when we are saying scan failed, this can mean scan found error and scan operation itself failed. 
* scan process failure: we should consider this as service failure and get admins notified about this. and also fail the build and retry when needed.
* scan finds some error: we may show it as warning to user, but build should succeed in this case.

Here in the exception handling, I believe we are looking for the first case, so I would say we have to device a plan to get ourselves notified and do exception handling properly, and fail in major cases. 
HDYT @navidshaikh @dharmit

Reply:
IMO, we should not fail the build if scan fails. The major offering from our service is building containers while scanning is an add-on feature which doesn't have to do with container build.

I am trying to understand why we want to fail the build if container is built and scan stage fails.

Reply:
@bamachrn thanks for putting it our more clearly. We should stop the execution and consider the build process failed if the scan process fails. If scanner finds some error/issues in the image, we note that and notify that to the user - the overall build job should be consider successful.

@navidshaikh 

> The major offering from our service is building containers while scanning is an add-on feature which doesn't have to do with container build.

We talk of our service as one that provides features beyond *just* building a container image. That's something that all services do for free. But the value addition we provide is linting, scanning, weekly scanning and repo tracking.